### PR TITLE
[Review] Update cuSignal documentation and Conda install ymls to better support Jetson Devices and prune dependencies

### PR DESCRIPTION
**Pull request by awthomp**
This should better reflect the true core dependencies of cuSignal by removing ecosystem dependencies like cuDF, cuML, and PyTorch. I will submit an additional yml including these dependencies for a 'fat' installation.

Ideally, this pruned down Conda yml will allow more OS and platforms to use Conda as the base installation mechanism. I would appreciate help testing, particularly on Windows.

### CODE REVIEW

**Comment by awthomp** [comment]
@beckernick -- we had discussed pruning the dependencies particularly in reference to the Windows config. Mind taking a look?

My 2020 resolution is to minimize merging my own PRs 😄

File: cusignal_conda_env.yml [hunk 1]
```diff
@@ -1,25 +1,17 @@
...
   - scipy
   - numpy
-  - pandas
-  - cudatoolkit=10.0
+  - cudatoolkit>=9.2,<10.2
   - boost
-  - jupyterlab
   - matplotlib
-  - nodejs
   - numba
```
Comment (beckernick) [inline cusignal_conda_env.yml]: numba now occurs twice

File: cusignal_conda_env.yml [hunk 2]
```diff
@@ -1,25 +1,17 @@
...
   - pip
   - pytest
   - pip:
-    - git+https://github.com/cupy/cupy.git
+    - cupy
```
Comment (beckernick) [inline cusignal_conda_env.yml]: You should be able to use conda cupy now. Check the cuDF environment yaml I believe we did it there

Comment (beckernick) [reply cusignal_conda_env.yml]: https://anaconda.org/conda-forge/cupy

Comment (awthomp) [reply cusignal_conda_env.yml]: Great -- Thanks for the tip, @beckernick! I'll need to move the conda-forge channel up in priority, as a conda install cupy  gave me v6.0.0. I'll add version limits too.

File: cusignal_conda_env.yml (hunk 1 above)
Comment (awthomp) [reply cusignal_conda_env.yml]: One numba is in the channels and another numba is in the dependencies. The Numba Jetson TX2 install says to use: `conda install -c numba numba`.

Comment (beckernick) [reply cusignal_conda_env.yml]: Ah, nice. Missed that

**Comment by awthomp** [comment]
Tests pass on my setup. Will check during the morning of 1/9 on the Jetson TX2. @evanmayer, do you mind seeing if this yml file update works on your Windows config and the cuSignal tests pass?

You should be able to run:
```python
conda env create -f cusignal_conda_base_env.yml
python setup.py install
pytest
```

Without errors on both Windows and (HOPEFULLY!) the Jetson devices. I'll submit a separate PR to update the documentation described in https://github.com/rapidsai/cusignal/issues/8

**Comment by awthomp** [comment]
This PR grew a bit and now can resolve https://github.com/rapidsai/cusignal/issues/8. I've also restructured the cusignal conda code to better reflect the other RAPIDS libraries and updated the `README.md` for clarity and to reflect the yml location changes.

**Comment by evanmayer** [comment]
Happy to test Windows, @awthomp:
`conda env create -f conda\environments\cusignal_base.yml` 
in the checkout dir gives:
~~~
ResolvePackageNotFound:
  - cupy[version='>=6.2.0']
~~~
Could this happen because Anaconda's Win64 Cupy package is only [6.0.0](https://anaconda.org/anaconda/cupy)? To investigate, I relaxed the version restriction to 6.0.0. That allowed the environment to solve, but of course had repercussions in the unit tests (<49% pass, with Cupy errors as culprits).

To work around this, I was then able to run `pip install cupy-cudaXXX` as in the previous Windows install process and pass the unit tests (123 passed, 54 warnings). I don't know if this means I have an extra 6.0.0 version of Cupy floating around in this env's `site-packages` dir now from the .yml install process, but this step apparently resolves that last dependency. It was easier than the first way, maybe still not perfect though.

All together:
~~~
conda env create -f conda\environments\cusignal_base.yml
pip install cupy-cuda101
python setup.py install
~~~
Is what passes as things stand with CUDA 10.1 installed

**Comment by evanmayer** [comment]
For the record, `conda env create -f conda\environments\cusignal_full.yml` 
in the checkout dir gives:
~~~
ResolvePackageNotFound:
  - cuml
  - cupy[version='>=6.2.0']
  - cugraph
  - cudf
~~~
Probably because those other conda packages aren't tagged for Windows support, but I'm guessing that's not a surprise.

**Comment by awthomp** [comment]
@evanmayer Thanks so much for testing! One more ask: could you please try the cusignal_base_jetson.yml file on Windows? This installs CuPy via pip versus Conda and *should* work on Windows. If so, we can either rename this yml file to reflect that it works on Windows and Jetson devices.

I know there's some hemming and hawing over conda vs pip for cupy, but from my experience, the pip install takes a really long time and I prefer the conda dependency solver.

**Comment by awthomp** [comment]
We're going to start refactoring the cuSignal structure to apply GPU CI/CD in preparation of nightly builds, Docker container generation, and submission to our conda channel. I'm going to merge the PR as the README is valid for each respective install. For Windows users, use the Jetson yml or build the environment as defined in the documentation.

**Pull request merged**